### PR TITLE
br: add summary to stream restore

### LINKED ISSUE #35236: Add summary to br restore point command
## Enhancement

Add summary to br restore point command

### PR DESCRIPTION
**Pull request by WangLe1321**
### What problem does this PR solve?

BR restore point command has no summary. User don't know whether it finished successfully.

Issue Number: close #35236

Problem Summary: BR restore point command has no summary.

### What is changed and how it works?

Add summary to tell user success or failed

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

1. exec restore point with full backup successfully

restore_success

2. exec restore point without full backup successfully

restore_point_success

3. exec restore point without full backup failed

restore_point_failed

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by joccau** [comment]
It is a simple summary currently. We need a summary containing more useful field about `restore point` later.

**Review by joccau — APPROVED** [review]
LGTM

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -932,7 +932,17 @@ func RunStreamRestore(
 	g glue.Glue,
 	cmdName string,
 	cfg *RestoreConfig,
-) error {
+) (err error) {
+	startTime := time.Now()
+	defer func() {
+		dur := time.Since(startTime)
+		if err != nil {
+			summary.Log(cmdName+" failed summary", zap.Error(err))
```
Comment (3pointer) [inline br/pkg/task/stream.go:940]: when using restore point with full backup storage. will this print two summary log?

Comment (WangLe1321) [reply br/pkg/task/stream.go:940]: The summary log here will not print twice repeatedly, but will print both full restore log and point restore log.

**Review by 3pointer — APPROVED** [review]
LGTM

**Comment by hawkingrei** [comment]
Wait for fixing an unstable test to merge.

**Pull request merged**